mobile/mobile-core/devices-service.ts

```diff
@@ -411,8 +411,15 @@ export class DevicesService implements Mobile.IDevicesService {
 	}
 
 	@exportedPromise("devicesService")
-	public getDebuggableApps(deviceIdentifier: string): IFuture<Mobile.IAndroidApplicationInformation[]> {
-		return this.$androidProcessService.getDebuggableApps(deviceIdentifier);
+	public getDebuggableApps(deviceIdentifiers: string[]): IFuture<Mobile.IAndroidApplicationInformation[]>[] {
+		return _.map(deviceIdentifiers, (deviceIdentifier: string) => this.getProcessService(deviceIdentifier).getDebuggableApps(deviceIdentifier));
+	}
+
+	private getProcessService(deviceIdentifier: string): { getDebuggableApps: (deviceIdentifier: string) => IFuture<Mobile.IAndroidApplicationInformation[]> } {
```

Comment:
I suggest skipping this check and use the `getDebuggableApps` from device's applicationManager:
```TypeScript
@exportedPromise("devicesService")
	public getDebuggableApps(deviceIdentifiers: string[]): IFuture<Mobile.IAndroidApplicationInformation[]>[] {
		return _.map(deviceIdentifiers, (deviceIdentifier: string) => this.getDebuggableAppsCore(deviceIdentifier));
	}

	private getDebuggableAppsCore(deviceIdentifier): IFuture<Mobile.IAndroidApplicationInformation[]> {
		return ((): Mobile.IAndroidApplicationInformation[] => {
			let device = this.getDeviceByIdentifier(deviceIdentifier);
			return device.applicationManager.getDebuggableApps().wait();
		}).future<Mobile.IAndroidApplicationInformation[]>()();
	}
```